packages/medusa/src/models/order-edit.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  BeforeInsert,
+  Column,
+  CreateDateColumn,
+  JoinColumn,
+  ManyToOne,
+  OneToMany,
+} from "typeorm"
+
+import { FeatureFlagEntity } from "../utils/feature-flag-decorators"
+import { resolveDbType } from "../utils/db-aware-column"
+import { OrderItemChange } from "./order-item-change"
+import { SoftDeletableEntity } from "../interfaces"
+import { generateEntityId } from "../utils"
+import { LineItem } from "./line-item"
+import { Order } from "./order"
+import { User } from "./user"
+
+@FeatureFlagEntity("order_editing")
+export class OrderEdit extends SoftDeletableEntity {
+  @Column()
+  order_id: string
+
+  @ManyToOne(() => Order, (o) => o.edits)
+  @JoinColumn({ name: "order_id" })
+  order: Order
+
+  @OneToMany(() => OrderItemChange, (oic) => oic.order_edit, {
+    eager: true,
+    cascade: true,
+  })
+  changes: OrderItemChange[]
+
+  @Column({ nullable: true })
+  internal_note?: string
+
+  @Column()
+  created_by: string
+
+  @ManyToOne(() => User)
+  @JoinColumn({ name: "created_by" })
+  created_by_user: User
+
+  @CreateDateColumn({ type: resolveDbType("timestamptz") })
+  created_at: Date
+
+  @Column({ nullable: true })
+  requested_by?: string
+
+  @ManyToOne(() => User)
+  @JoinColumn({ name: "requested_by" })
+  requested_by_user: User
+
+  @CreateDateColumn({ type: resolveDbType("timestamptz"), nullable: true })
+  requested_at?: Date
```

Comment:
I didn’t mean to not set it in the decorator.  But the problem is that If we manipulate the property you can’t set null even if the field is nullable. Because of the type error you get. Let it that way, but each time we create a new entity i mentioned that point ^^. Also, it is not a pattern since the cart, address, shipping etc specify that null possibility if you look at it.